server/webhook.go

```diff
@@ -5,44 +5,75 @@ package server
 
 import (
 	"bytes"
+	"context"
 	"encoding/json"
-	"io"
+	"fmt"
 	"io/ioutil"
 	"net/http"
-
-	"github.com/pkg/errors"
 )
 
+type WebhookValidationError struct {
+	err string
+}
+
+func (e *WebhookValidationError) Error() string {
+	return fmt.Sprintf("%v", e.err)
+}
+
```

Comment:
This implements only partially what I suggested. This just wraps the error with a type, and does not do anything else. As it is, it's not very useful. I was suggesting that a field added to the struct which will indicate what is not set- url/username/text.

Please also add godocs to both the struct and the method.

Reply:
Can you please send me an example for what you mean by this?

Reply:
Sure.

```go
// WebhookValidationError contains an error in the webhook payload.
type WebhookValidationError struct {
 field string
}

// Error implements the error interface.
func (e *WebhookValidationError) Error() string {
	return "invalid" + e.field
}

func newWebhookValidationError(field string) *WebhookValidationError {
  return &WebhookValidationError{
   field: field,
  }
}

// from code
if webhookURL == "" {
	return newWebhookValidationError("webhook URL")
}

// in test
var whError *WebhookValidationError
require.True(errors.As(err, &whError))
assert.Equal(whError.field, "webhook URL")
```

Let me know if that helps. (Code is untested, I just wrote here directly)

Reply:
This helps immensely! Thank you so much!